core/deployment/src/main/java/io/quarkus/deployment/steps/RegisterForReflectionBuildStep.java

```diff
@@ -36,12 +35,6 @@ public void build(BuildProducer<ReflectiveClassBuildItem> reflectiveClass) {
             AnnotationValue classNamesValue = i.value("classNames");
 
             ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
-            if (targetsValue == null && classNamesValue == null) {
-                ClassInfo classInfo = i.target().asClass();
-                registerClass(classLoader, classInfo.name().toString(), methods, fields, ignoreNested, serialization,
-                        reflectiveClass);
-                continue;
-            }
```

Comment:
Why this change? When `RegisterForReflection` is on a class without any of these attributes defined, the class itself should be registered for reflection.

Reply:
You are right it's undue change. Already reverted. Sorry for that.